quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -223,131 +227,135 @@ async fn index_metadata(
 /// Extends an existing SQL string with the generated filter range appended to the query.
 ///
 /// This method is **not** SQL injection proof and should not be used with user-defined values.
-fn write_sql_filter<V: Display>(
-    sql: &mut String,
-    field_name: impl Display,
-    filter_range: &FilterRange<V>,
-    value_formatter: impl Fn(&V) -> String,
-) {
-    match &filter_range.start {
-        Bound::Included(value) => {
-            let _ = write!(sql, " AND {} >= {}", field_name, (value_formatter)(value));
-        }
-        Bound::Excluded(value) => {
-            let _ = write!(sql, " AND {} > {}", field_name, (value_formatter)(value));
-        }
-        Bound::Unbounded => {}
+fn write_sql_filter<'a, V: Display>(
+    mut sql: &'a mut SelectStatement,
+    field_name: Splits,
+    filter_range: &'a FilterRange<V>,
+    value_formatter: impl Fn(&V) -> Expr,
+) -> &'a mut SelectStatement {
+    if let Bound::Included(value) = &filter_range.start {
+        sql = sql.cond_where(Expr::col(field_name.clone()).gte((value_formatter)(value)))
```

Comment:
No need to reassign since the function is `&mut self`

```suggestion
        sql.cond_where(Expr::col(field_name.clone()).gte((value_formatter)(value)));
```